src/sentry/api/endpoints/organization_events_has_measurements.py

```diff
@@ -0,0 +1,118 @@
+from datetime import timedelta
+
+import sentry_sdk
+from django.core.cache import cache
+from django.utils import timezone
+from rest_framework import serializers
+from rest_framework.response import Response
+
+from sentry.api.bases import NoProjects, OrganizationEventsV2EndpointBase
+from sentry.snuba import discover
+from sentry.utils.hashlib import md5_text
+
+MEASUREMENT_TYPES = {
+    "web": [
+        "measurements.fp",
+        "measurements.fcp",
+        "measurements.lcp",
+        "measurements.fid",
+        "measurements.cls",
+    ],
+    "mobile": [
+        "measurements.app_start_cold",
+        "measurements.app_start_warm",
+        "measurements.frames_total",
+        "measurements.frames_slow",
+        "measurements.frames_frozen",
+        "measurements.stall_count",
+        "measurements.stall_stall_total_time",
+        "measurements.stall_stall_longest_time",
+    ],
+}
+
+
+class EventsHasMeasurementsQuerySerializer(serializers.Serializer):
+    transaction = serializers.CharField(max_length=200)
+    type = serializers.ChoiceField(choices=list(MEASUREMENT_TYPES.keys()))
+
+    def validate(self, data):
+        # only allow one project at a time in order to cache the results
+        # for a unique transaction
+        project_ids = self.context.get("params", {}).get("project_id", [])
+        if len(project_ids) != 1:
+            raise serializers.ValidationError("Only 1 project allowed.")
+        return data
+
+
+class OrganizationEventsHasMeasurementsEndpoint(OrganizationEventsV2EndpointBase):
+    def get_snuba_params(self, request, organization, check_global_views=True):
```

Comment:
Does this endpoint need global-views? We could validate that they only pass a single project first then get snuba_params right

Reply:
Actually, good point, and I already validate they only passed a single project in the serializer above so the global views check can be safely removed.